### PR TITLE
API Remove GraphQL

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-admin/issues/1799

### CODE REVIEW

File: code/LeftAndMain.php [hunk 1]
```diff
@@ -53,6 +53,7 @@
 use SilverStripe\View\ArrayData;
 use SilverStripe\View\Requirements;
 use SilverStripe\View\SSViewer;
+use stdClass;
```
Comment (ChloeSartorelli) [inline code/LeftAndMain.php]:
```suggestion
```
Doesn't seem to be used (if it is, please say which line it's on 'cause I missed it)

File: code/LeftAndMain.php [hunk 2]
```diff
@@ -429,12 +430,16 @@ public function schema(HTTPRequest $request): HTTPResponse
     /**
      * Creates a successful json response
      */
-    protected function jsonSuccess(int $statusCode, array $data = []): HTTPResponse
+    protected function jsonSuccess(int $statusCode, ?array $data = null): HTTPResponse
```
Comment (ChloeSartorelli) [inline code/LeftAndMain.php:443]: Why not just use empty array as empty value?

File: code/LeftAndMain.php [hunk 3]
```diff
@@ -429,12 +430,16 @@ public function schema(HTTPRequest $request): HTTPResponse
...
     {
         if ($statusCode < 200 || $statusCode >= 300) {
             throw new InvalidArgumentException("Status code $statusCode must be between 200 and 299");
         }
-        $body = empty($data) ? '' : json_encode($data, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+        if (is_null($data)) {
+            $body = '';
+        } else {
+            $body = json_encode($data, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
+        }
```
Comment (ChloeSartorelli) [inline code/LeftAndMain.php:452]: This change doesn't seem to actually do anything? Doesn't it give the exact same result it did before?

File: code/LeftAndMain.php [hunk 4]
```diff
@@ -479,6 +484,23 @@ public function jsonError($errorCode, $errorMessage = null)
...
+        $securityToken = SecurityToken::inst();
+        $name = $securityToken->getName();
+        if ($name === 'SecurityID') {
+            // SecurityID is a non-standard HTTP header, it should be prefixed with 'X-'
+            $name = "X-SecurityID";
+        }
+        $value = $request->getHeader($name);
+        if (!$securityToken->check($value)) {
+            $this->jsonError(400);
+        }
```
Comment (ChloeSartorelli) [inline code/LeftAndMain.php]: Why not just `SecurityToken::inst()->checkRequest($request)`? Seems like this whole method is unnecessary?

File: tests/php/LeftAndMainTest.php [hunk 5]
```diff
@@ -155,6 +155,11 @@ public function testCanView()
         $menuItems = $menuItems->column('Code');
         sort($menuItems);
 
+        // filter out any menus from other modules that may be installed
+        $menuItems = array_filter($menuItems, function ($item) {
+            return strpos($item, 'SilverStripe-Admin-') === 0;
+        });
```
Comment (ChloeSartorelli) [inline tests/php/LeftAndMainTest.php]: Why is this needed now when it wasn't before?

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: code/LeftAndMain.php (hunk 1 above)
Comment (emteknetnz) [reply code/LeftAndMain.php]: Removed

File: code/LeftAndMain.php (hunk 2 above)
Comment (emteknetnz) [reply code/LeftAndMain.php:443]: Sometimes you actually want to return an empty JSON array

File: code/LeftAndMain.php (hunk 3 above)
Comment (emteknetnz) [reply code/LeftAndMain.php:452]: It's to allow JSON encoding an empty array. Previously it would return an empty body instead.

File: code/LeftAndMain.php (hunk 4 above)
Comment (emteknetnz) [reply code/LeftAndMain.php]: That is correct. Will update.

File: tests/php/LeftAndMainTest.php (hunk 5 above)
Comment (emteknetnz) [reply tests/php/LeftAndMainTest.php]: No longer required I think, will remove

File: code/LeftAndMain.php (hunk 2 above)
Comment (ChloeSartorelli) [reply code/LeftAndMain.php:443]: Discussed in person - the reason reason is so that you can have an empty body, which seems sensible to allow.

File: code/LeftAndMain.php (hunk 3 above)
Comment (ChloeSartorelli) [reply code/LeftAndMain.php:452]: See https://github.com/silverstripe/silverstripe-admin/pull/1802#discussion_r1726207432

**Review by GuySartorelli — APPROVED** [review]
LGTM, will merge when the other PRs are ready

**Review by GuySartorelli — APPROVED** [review]
re-approved most recent changes

**Review by GuySartorelli — APPROVED** [review]
re-approved most recent changes

**Comment by GuySartorelli** [comment]
Needs rebase

**Comment by emteknetnz** [comment]
Done

**Review by GuySartorelli — APPROVED** [review]
LGTM
